### PR TITLE
ScaffoldMessenger Migration Guide

### PR DESCRIPTION
**Pull request by Piinks**
Migration guide for change: https://github.com/flutter/flutter/pull/64101
Change with framework migrations: https://github.com/flutter/flutter/pull/64170
Issue: https://github.com/flutter/flutter/issues/62921

### CODE REVIEW

File: src/docs/release/breaking-changes/index.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ release, and listed in alphabetical order:
 * [Android v1 embedding app and plugin creation deprecation][]
 * [Network Policy on iOS and Android][]
 * [The new Form, FormField auto-validation API][]
+* [SnackBars managed by the ScaffoldMessenger]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/index.md:51]:
```suggestion
* [SnackBars managed by the ScaffoldMessenger][]
```

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+scaffoldMessengerKey.currentState.hideCurrentSnackBar(mySnackBar);
+scaffoldMessengerKey.currentState.removeCurrentSnackBar(mySnackBar);
+
+// The root ScaffoldMessenger can also be accessed by providing a key to 
+// MaterialApp.scaffoldMessengerKey. This way, the ScaffoldMessengerState can be directly accessed
+// without first obtaining it from a BuildContext via ScaffoldMessenger.of. From the key, use
+// the GlobalKey.currentState getter.
+final GlobalKey<ScaffoldMessengerState> rootScaffoldMessengerKey = GlobalKey<ScaffoldMessengerState>();
+MaterialApp(
+  scaffoldMessengerKey: rootScaffoldMessengerKey,
```
Comment (Piinks) [inline src/docs/release/breaking-changes/scaffold-messenger.md:143]:
```suggestion
  scaffoldMessengerKey: rootScaffoldMessengerKey,
  home: ...
```

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+rootScaffoldMessengerKey.currentState.removeCurrentSnackBar(mySnackBar);
+```
+
+## Timeline
+
+Landed in version: TBD<br>
+In stable release: not yet
+
+The previous `ScaffoldState` methods (`showSnackBar`, `hideCurrentSnackBar`, and
+`removeCurrentSnackBar`) are deprecated TBD. Calling on these methods will divert the `SnackBar` to
```
Comment (Piinks) [inline src/docs/release/breaking-changes/scaffold-messenger.md:157]:
```suggestion
`removeCurrentSnackBar`) are deprecated. Calling on these methods will divert the `SnackBar` to
```

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+the `ScaffoldMessenger`.
+
+## References
+
+{% include master-api.md %}
+
+Design doc:
+* [Scrollable `AlertDialog`][]
+
```
Comment (Piinks) [inline src/docs/release/breaking-changes/scaffold-messenger.md:166]:
```suggestion
```

**Comment by sfshaza2** [comment]
Sorry, @Piinks, but I just landed a largish PR that updates the breaking change files. Please fix the conflicts at your convenience. :) (As it hasn't landed, I see we are in no rush.)

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+          Scaffold.of(context).showSnackBar(SnackBar(
+            content: const Text('snack'),
+            duration: const Duration(seconds: 1),
+            action: SnackBarAction(
+              label: 'ACTION',
+              onPressed: () { },
+            ),
+          ));
+        },
+        child: const Text('SHOW SNACK'),
```
Comment (domesticmouse) [inline src/docs/release/breaking-changes/scaffold-messenger.md:47]: Any particular reason to use all caps here and above? I'm guessing your intent is to make it obvious that the developer can change these, but it just feels shouty... =)

Comment (Piinks) [reply src/docs/release/breaking-changes/scaffold-messenger.md:47]: Oh I was just copying a sample from the API docs. 😜

**Comment by Piinks** [comment]
Update: @sfshaza2 this is still WIP, the change was reverted so we don't have a build tag yet. :)

File: src/docs/release/breaking-changes/scaffold-messenger.md (hunk 5 above)
Comment (domesticmouse) [reply src/docs/release/breaking-changes/scaffold-messenger.md:47]: Fair 😋

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+)
+
+rootScaffoldMessengerKey.currentState.showSnackBar(mySnackBar);
+rootScaffoldMessengerKey.currentState.hideCurrentSnackBar(mySnackBar);
+rootScaffoldMessengerKey.currentState.removeCurrentSnackBar(mySnackBar);
+```
+
+## Timeline
+
+Landed in version: TBD<br>
```
Comment (Piinks) [inline src/docs/release/breaking-changes/scaffold-messenger.md:154]:
```suggestion
Landed in version: 1.23.0-13.0.pre
```

**Comment by Piinks** [comment]
@Sfshaza we finally have a build tag for this one, ready to go with your approval! 🎉

**Comment by Hixie** [comment]
Please include the error message that you're likely to see if you haven't migrated, so that people can find this by Googling the error message.

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+          onPressed: () { },
+        ),
+      ));
+    },
+    child: const Text('SHOW SNACK'),
+  ),
+);
+```
+
+When presenting a `SnackBar` during a transition, the `SnackBar` will `Hero`, animating smoothly to
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/scaffold-messenger.md:77]: Never heard "Hero" as a verb. Is this a common usage?

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+the next page.
+
+The `ScaffoldMessenger` creates a scope in which all descendant `Scaffold`s will register to receive
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/scaffold-messenger.md:80]: Delete "will".

File: src/docs/release/breaking-changes/scaffold-messenger.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+`SnackBar`s, which is how they persist across these transitions. When using the root
+`ScaffoldMessenger` provided by the `MaterialApp`, all descendant `Scaffold`s will receive `SnackBar`s,
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/scaffold-messenger.md:82]: Delete "will".

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
A couple minor nits but, overall, looks great!

**Review by sfshaza2 — APPROVED** [review]
LGTM

**Pull request merged**